INSTALLATION AND DATA IMPORTATION

In [3]:
#Installing and importing libraries for data importation
!pip install pandas



[notice] A new release of pip is available: 24.1.2 -> 24.2
[notice] To update, run: pip install --upgrade pip


In [4]:
#import libraries
import pandas as pd

In [6]:
#import dataset excel
pd.read_excel("diamonds_new.xlsx")


price  carat        cut color clarity  depth  table     x     y     z
0        326   0.23      Ideal     E     SI2   61.5   55.0  3.95  3.98  2.43
1        326   0.21    Premium     E     SI1   59.8   61.0  3.89  3.84  2.31
2        327   0.23       Good     E     VS1   56.9   65.0  4.05  4.07  2.31
3        334   0.29    Premium     I     VS2   62.4   58.0  4.20  4.23  2.63
4        335   0.31       Good     J     SI2   63.3   58.0  4.34  4.35  2.75
...      ...    ...        ...   ...     ...    ...    ...   ...   ...   ...
53915   2757   0.72      Ideal     D     SI1   60.8   57.0  5.75  5.76  3.50
53916   2757   0.72       Good     D     SI1   63.1   55.0  5.69  5.75  3.61
53917   2757   0.70  Very Good     D     SI1   62.8   60.0  5.66  5.68  3.56
53918   2757   0.86    Premium     H     SI2   61.0   58.0  6.15  6.12  3.74
53919   2757   0.75      Ideal     D     SI2   62.2   55.0  5.83  5.87  3.64

[53920 rows x 10 columns]

In [5]:
#import the dataset as a pandas dataframe
class1=pd.read_excel("diamonds_new.xlsx")

#View the new dataframe called class1
class1

price  carat        cut color clarity  depth  table     x     y     z
0        326   0.23      Ideal     E     SI2   61.5   55.0  3.95  3.98  2.43
1        326   0.21    Premium     E     SI1   59.8   61.0  3.89  3.84  2.31
2        327   0.23       Good     E     VS1   56.9   65.0  4.05  4.07  2.31
3        334   0.29    Premium     I     VS2   62.4   58.0  4.20  4.23  2.63
4        335   0.31       Good     J     SI2   63.3   58.0  4.34  4.35  2.75
...      ...    ...        ...   ...     ...    ...    ...   ...   ...   ...
53915   2757   0.72      Ideal     D     SI1   60.8   57.0  5.75  5.76  3.50
53916   2757   0.72       Good     D     SI1   63.1   55.0  5.69  5.75  3.61
53917   2757   0.70  Very Good     D     SI1   62.8   60.0  5.66  5.68  3.56
53918   2757   0.86    Premium     H     SI2   61.0   58.0  6.15  6.12  3.74
53919   2757   0.75      Ideal     D     SI2   62.2   55.0  5.83  5.87  3.64

[53920 rows x 10 columns]

In [5]:
#Check the further information about dataset
class1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53920 entries, 0 to 53919
Data columns (total 10 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   price    53920 non-null  int64  
 1   carat    53920 non-null  float64
 2   cut      53920 non-null  object 
 3   color    53920 non-null  object 
 4   clarity  53920 non-null  object 
 5   depth    53920 non-null  float64
 6   table    53920 non-null  float64
 7   x        53920 non-null  float64
 8   y        53920 non-null  float64
 9   z        53920 non-null  float64
dtypes: float64(6), int64(1), object(3)
memory usage: 4.1+ MB


In [7]:
class1.describe().transpose

<bound method DataFrame.transpose of               price         carat         depth         table             x  \
count  53920.000000  53920.000000  53920.000000  53920.000000  53920.000000   
mean    3933.492174      0.798018     61.749560     57.456769      5.731337   
std     3989.963677      0.474061      1.432414      2.233927      1.121807   
min      326.000000      0.200000     43.000000     43.000000      0.000000   
25%      950.000000      0.400000     61.000000     56.000000      4.710000   
50%     2401.000000      0.700000     61.800000     57.000000      5.700000   
75%     5327.000000      1.040000     62.500000     59.000000      6.540000   
max    18823.000000      5.010000     79.000000     95.000000     10.740000   

                  y             z  
count  53920.000000  53920.000000  
mean       5.734703      3.538854  
std        1.142192      0.705736  
min        0.000000      0.000000  
25%        4.720000      2.910000  
50%        5.710000      3.530000  

## PROBABILITY OF DIAMONDS DATASET ##

**A. Basic Concepts of Probability**

In [1]:
# Define a sample space of tossing a die
#A die has 6 faces
sample_space = [1, 2, 3, 4, 5, 6]

In [2]:
# Define an event 
event = [1, 2, 3]

In [3]:
# Calculate the probability of the event
probability = len(event) / len(sample_space)
print("Probability of the events occuring is:", probability)

Probability of the events occuring is: 0.5


The  probability of 3 outcomes in an event is 0.5 or 50%

**ii) Probability of a categorical variable**

Using the diamonds dataset, calculate the probability of getting a "Premium" cut diamond.

In [3]:
#First we get the different number cuts of diamonds in the dataset (This is the sample space)
cut_total = class1['cut'].count()
#View the result
cut_total

53920

In [4]:
#Then get the count of just "premium" diamonds (This is the event)
premium = class1[class1['cut'] == "Premium"]['cut'].count()
#View result
premium

13789

In [5]:
#Now calculate the probability of having a Premium diamond
probability=premium/cut_total
#View result
probability

0.2557307121661721

The probability is 25.57%

What is the probability for the colour J?

In [17]:
no_J = class1[class1['color'] == "J"]['color'].count()
#View results
no_J

2807

In [12]:
#Probability of total color
color_total = class1['color'].count()
#View result
color_total


53920

In [13]:
#Calculate probability of J
prob_J = no_J/color_total
#View result
prob_J

0.05205860534124629

The probability of color J is 0.052 or 5.2%

**Probability of a continuous variable**

**B. Types of probability**

**i)Conditional Probability**

Conditional probability is  the probability that depends on a previous result or event, calculated using the formula:

P(A | B) = P(A and B) / P(B), where A and B are two events

P(A∩B) = the probability that event A and event B both occur. 

P(B) = the probability that event B occurs.

Example 1: In a survey among 1000 students, 70% play football, 60% play basketball, and 40% play both sports. If a student is chosen at random and it is known that the student plays basketball, what is the probability that the student also plays football?

There are 1000 students in the survey.

Number of students who play football = n(A) = 0.7*1000 = 700

Number of students who play basketball = n(B) = 0.6*1000 = 600

Number of students who play both sports = n(A ∩ B) = 40 = 0.4*1000 = 400

To find the probability that a student plays football given that they play basketball, we use the conditional probability formula:

P(A|B) = n(A ∩ B) / n(B)

Substituting the values, we get:

P(A|B) = 400 / 600 = 2/3 which is 0.66

Therefore, probability that a randomly chosen student who plays basketball also plays football is 2/3.

In [9]:
#The above in python would be
nA = 0.7*1000
nB = 0.6*1000
AnB = 0.4*1000
#Finding probability
probability3 = AnB/nB
#View result
print("Probability that a student who plays basketball also plays football when chosen at random is:", probability3)

Probability that a student who plays basketball also plays football when chosen at random is: 0.6666666666666666


Example 2: What is the probability of having a Premium diamond that is "J" coloured?

In this example, the event A is Premium diamond, and event B is J colour.

We already have these values nA=premium= 13789

nB=no_J= 2807

We don't have the n(AnB) which we need to find

Remember P(A)*P(B)= P(Premium)*P(J)= 0.0133


In [14]:
#First filter the  data for only premium diamonds that are J coloured
filtered_data = (class1['cut'] == 'Premium') & (class1['color'] == 'J')

In [15]:
#number of outcomes of selecting Premium and J (This is the n(AnB))
premium_j_count = class1[filtered_data]["cut"].count()
#View result
premium_j_count

808

In [18]:
#Conditional probability
probability4 = premium_j_count/no_J
#View result as
print("Probability a diamond is premium and J in colour is:", probability4)

Probability a diamond is premium and J in colour is: 0.2878517990737442


**ii) Exercise**

a) Using the "Student_performance.xlsx" dataset, what is the probability of;

i)A student studying Information_Technology?

ii) A female student studying Computer_Science?

iii) A male student studying Data_Science?